### PR TITLE
Refactor dockerfiles & build process, fix tests

### PR DESCRIPTION
**Pull request by jfchevrette**
Changes:
- Use supported base images (ubi8, quay.io)
- Refactor Dockerfiles & Makefile to use multistage builds
- fix tests which were failing (and had been for a long time probably) because of changes in schemas made many months ago
- Update vault version used in tests to match the one we use in prod as the old one has incompatibilities with data formats

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,10 @@
-FROM alpine:3.7
-RUN apk add --no-cache ca-certificates
-RUN adduser -D -g '' vault-manager
-USER vault-manager
-COPY vault-manager query.graphql /
+FROM registry.access.redhat.com/ubi8/go-toolset:latest as builder
```
Comment (rporres) [inline Dockerfile:1]: Can we use a version here instead of latest?

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,10 @@
...
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go test ./...
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo ./cmd/vault-manager
```
Comment (rporres) [inline Dockerfile:4]: why not making this a `Makefile` target?

File: tests/Dockerfile.tests [hunk 3]
```diff
@@ -1,5 +1,9 @@
-FROM centos:7
+FROM registry.access.redhat.com/ubi8/go-toolset:latest as builder
+COPY . .
+RUN CGO_ENABLED=0 GOOS=linux go test ./...
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo ./cmd/vault-manager
```
Comment (rporres) [inline tests/Dockerfile.tests:4]: why not making these commands `Makefile` targets?

**Review by rporres — COMMENTED** [review]
Commented mainly in the build parts... Unfortunately I don't have enough knowledge of the project to review properly the rest

File: Dockerfile (hunk 1 above)
Comment (jfchevrette) [reply Dockerfile]: yes we should! fixed!

File: Dockerfile (hunk 2 above)
Comment (jfchevrette) [reply Dockerfile:4]: Good suggestion. Feel free to send a PR :)

File: tests/Dockerfile.tests (hunk 3 above)
Comment (jfchevrette) [reply tests/Dockerfile.tests:4]: Good suggestion. Feel free to send a PR :)

**Pull request merged**